caikit_nlp/modules/text_embedding/crossencoder.py

```diff
@@ -0,0 +1,807 @@
+            else {}
+        )
+
+    @TokenizationTask.taskmethod()
+    def run_tokenizer(
+        self,
+        text: str,
+    ) -> TokenizationResults:
+        """Run tokenization task against the model
+
+        Args:
+            text: str
+                Text to tokenize
+        Returns:
+            TokenizationResults
+                The token count
+        """
+        result = self.model.get_tokenized([text], return_offsets_mapping=True)
+
+        mapping = [
+            interv for interv in result.offset_mapping[0] if (interv[1] - interv[0]) > 0
+        ]
+        tokens = [Token(start=i[0], end=i[1], text=text[i[0] : i[1]]) for i in mapping]
+
+        return TokenizationResults(token_count=len(result.input_ids[0]), results=tokens)
+
+    @classmethod
+    def _get_ipex(cls, ipex_flag):
+        """Get IPEX optimization library if enabled and available, else return False
+
+        Returns ipex library or False
+        """
+        ret = False
+
+        # Enabled by environment variable
+        # When IPEX is not false, attempt to import the library and use it.
+        if ipex_flag:
+            try:
+                ret = importlib.import_module("intel_extension_for_pytorch")
+            except Exception as ie:  # pylint: disable=broad-exception-caught
+                # We don't require the module so catch, log, proceed to return False
+                msg = (
+                    f"IPEX enabled in env, but skipping ipex.optimize() because "
+                    f"import intel_extension_for_pytorch failed with exception: {ie}"
+                )
+                logger.warning(msg, exc_info=True)
+
+        return ret
+
+    @staticmethod
+    def _select_device(use_ipex, device):
+        """Use environment variables and availability to determine the device to use"""
+        if use_ipex:
+            # If enabled, use "xpu" (IPEX on GPU instead of IPEX on CPU)
+            if device == "xpu":
+                return "xpu"
+        elif device == "mps" and mps.is_built() and mps.is_available():
+            # Never use on ipex, but otherwise use mps if enabled and available
+            return "mps"
+
+        return "cuda" if torch.cuda.is_available() else None
+
+    @staticmethod
+    def _get_backend(use_ipex, use_device):
+        """Determine the backend to use for torch compile.
+
+        Considers global ipex if enabled first, next mps device, finally defaults.
+
+        Returns the backend for torch.compile()
+        """
+        if use_ipex:
+            return "ipex"
+        if use_device == "mps":
+            return mps
+        return "inductor"  # default backend
+
+    @staticmethod
+    def _optimize(model, ipex, device, autocast, pt2_compile):
+        if ipex:
+            if autocast:  # IPEX performs best with autocast using bfloat16
+                model = ipex.optimize(
+                    model, dtype=torch.bfloat16, weights_prepack=False
```

Comment:
Took this out because we won't really properly test the ipex options for short-term cross-encoder needs.  But FYI, various config names related to dtype/bfloat16 were rejected as confusing with other uses so for embeddings it ended up ipex + autocast is how you take advantage ipex with bloat16 speed.